docker/Dockerfile

```diff
@@ -2,36 +2,48 @@
 # Install OpenSearch Benchmark to build a Docker image #
 ########################################################
 
-ARG VERSION
-ARG BUILD_ENV=production
+#
+# Stage 1: build packages and compile where needed
```

Comment:
did we get any significant size reduction in final image by splitting into multi-stage. It is fairly simple dockerfile.

Reply:
See this comment:
https://github.com/opensearch-project/opensearch-benchmark/pull/632#discussion_r1739523746

Reply:
Goes down from about 700 MB to 500 MB with two stages.